File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        map<int, int> map;
        for (int x : nums) {
            map[x]++;
            if (map[x]> 1) {
                return true;
            }
            
        }
        return false;
    }
};

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        hi, lo = len(nums) -  1, 0


        while hi > lo:

            mid = (hi + lo) // 2

            if nums[mid] >= nums[hi]:
                lo = mid + 1
            else:
                hi = mid 

        return nums[lo]